qutip/tests/core/test_qobjevo.py

```diff
@@ -121,376 +149,234 @@ def _assert_qobjevo_different(obj1, obj2):
     assert any(obj1(t) != obj2(t) for t in np.random.rand(10) * .9 + 0.05)
 
 
-def _add(a, b):
-    return a + b
-
-def _sub(a, b):
-    return a - b
-
-def _mul(a, b):
-    return a * b
-
 def _div(a, b):
     return a / b
 
-def _matmul(a, b):
-    return a @ b
-
-def _tensor(a, b):
-    return a & b
 
-def _conj(a):
-    return a.conj()
-
-def _dag(a):
-    return a.dag()
-
-def _trans(a):
-    return a.trans()
-
-def _neg(a):
-    return -a
+class TestQobjevo:
+    def test_call(self, pseudo_qevo, coeff_type):
+        # test creation of QobjEvo and call
+        qevo = QobjEvo(*getattr(pseudo_qevo, coeff_type)())
+        assert isinstance(qevo(0), Qobj)
+        assert qevo.isoper
+        assert not qevo.const
+        assert not qevo.issuper
+        _assert_qobjevo_equivalent(pseudo_qevo, qevo)
+
+    @pytest.mark.parametrize('coeff_type',
+                             ['func_coeff', 'string',
+                              'spline', 'array', 'logarray'])
```

Comment:
Why re-parametrise `coeff_type` with the same values that are already defined?

Reply:
To remove one type: `func_call`.  `skip` sound wrong since we would be skipping a test that should not exist. If there is a feature in pytest that skip a param without listing it in the test list this is what should be used.

Reply:
Ah, I missed that one of them was missing.  I thought it was the same list.  How come we don't need to test that one in this case?

Reply:
This is for the creation of `QobjEvo` with `Qobj * Coefficient`. Function returning a Qobj cannot be created this way.

Reply:
Ok cool, sounds sensible.